docs/development/versioning.md

```diff
@@ -0,0 +1,74 @@
+# Versioning
+
+## Version changes
+
+Polars adheres to the [semantic versioning](https://semver.org/) specification.
+
+As Polars has not released its `1.0.0` version yet, breaking releases lead to a minor version increase (e.g. from `0.18.15` to `0.19.0`), while all other releases increment the patch version (e.g. from `0.18.14` to `0.18.15`)
+
+## Policy for breaking changes
+
+Polars takes backwards compatibility seriously, but we are not afraid to change things if it leads to a better product.
+
+!!! warning Rust users only
+
+    The Rust API for Polars is currently not considered stable.
```

Comment:
Can we word this less drastically? Maybe that we are more certain about the lazy API, but that the internals may expect more breaking API changes. Also because it is harder to hide the API than it is in python.

Reply:
Sure, will change the wording here!

Reply:
We don't really have to say anything about the stability of the API. We just need to mention that we don't deprecate things in Rust. I rewrote it to:

> Breaking changes for the Rust API are not deprecated first, but _will_ be listed in the changelog.
> Supporting deprecated functionality would slow down development too much at this point in time.